cotainr/pack.py

```diff
@@ -124,25 +155,89 @@ def _check_conda_bootstrap_integrity(self):
 
-    def _download_conda_installer(self, *, path):
+    def _display_miniforge_license_for_acceptance(self, *, installer_path):
+        """
+        Extract and display Miniforge installer license for acceptance.
+
+        Runs the Miniforge bootstrap installer to extract the license, displays
+        it to the user, and prompts for acceptance of the license terms. Exits
+        if the license terms are not accepted.
+
+        Parameters
+        ----------
+        installer_path : pathlib.Path
+            The path of the Miniforge installer to run to bootstrap Conda.
+
+        Raises
+        ------
+        RuntimeError
+            If unable to extract a license from the Miniforge installer.
+
+        Notes
+        -----
+        This assumes that the Miniforge installer, as it is run, prompts the
+        user for pressing ENTER to display the license, then displays the
+        license and prompts the user to answer "yes" to accept the license
+        terms.
+
+        We try to "forward" this flow to the user by extracting the text shown
+        when running the installer and pressing ENTER. We then prompt for a
+        "yes" to the license terms.
+        """
+        with subprocess.Popen(
+            ["bash", f"{installer_path.name}"],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            text=True,
+        ) as process:
+            license, _ = process.communicate(
+                # "press" ENTER to display the license and capture it
+                "\n"
+            )
+            process.kill()  # We only use this process to extract the license
+
+        if license:
+            license = license.replace(
+                # remove prompt for pressing enter (as we have already done this...)
+                "Please, press ENTER to continue\n>>> ",
+                "\n",
+            )
+            print(license)
+            val = input()  # prompt user for acceptance of license terms
+            if val != "yes":
+                print(
+                    "You have not accepted the Miniforge installer license. Aborting!"
+                )
+                sys.exit(0)
+
+            self.license_accepted = True
+            print("You have accepted the Miniforge installer license.")
+        else:
+            raise RuntimeError(
+                "No license seems to be displayed by the Miniforge installer."
+            )
+
+    def _download_miniforge_installer(self, *, installer_path):
         """
-        Download the Conda installer to `path`.
+        Download the Miniforge installer to `path`.
 
         Parameters
         ----------
-        path : pathlib.Path
+        installer_path : pathlib.Path
             The path to download the conda installer to.
         """
-        conda_installer_url = (
+        miniforge_installer_url = (
             "https://github.com/conda-forge/miniforge/releases/latest/download/"
             "Miniforge3-Linux-x86_64.sh"
```

Comment:
dont we support mac os x?

Reply:
Singularity and Apptainer only works on Linux, so we only support Linux. On Windows and Mac OS, you need a Linux VM to run Singularity and Apptainer.

https://apptainer.org/docs/admin/main/installation.html#installation-on-windows-or-mac
https://docs.sylabs.io/guides/3.0/user-guide/installation.html#install-on-windows-or-mac

Reply:
However, I guess, at some point we might need to support ARM architectures as well, not just x86_64.